src/reconcile/rec_hs.c

```diff
@@ -347,8 +349,15 @@ __rec_hs_cursor_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint32_t btr
      * If we find a key with a timestamp larger than or equal to the specified timestamp then the
      * specified timestamp must be mixed mode.
      */
-    WT_ASSERT_ALWAYS(
-      session, ts == 1 || ts == WT_TS_NONE, "out-of-order timestamp update detected");
+    char tp_string[4][WT_TS_INT_STRING_SIZE];
+    WT_ASSERT_ALWAYS(session, ts == 1 || ts == WT_TS_NONE,
+      "out-of-order timestamp update detected, found an existing update with key=%s, with "
+      "hs_start_ts=%s, start_ts=%s, stop_ts=%s, that exists later than specified ts=%s",
+      __wt_key_string(session, key->data, key->size, S2BT(session)->key_format, &key_string),
```

Comment:
I think it is not allowed to print out user key directly in production. It leaks user data and it is a security breach.

Reply:
Oh sorry about this, I'll remove it. Can we log the timestamp values?

Reply:
yes, logging timestamp values is fine.

Reply:
In debug, I think it is ok. @kommiharibabu ?

Reply:
I don't think that we can find any information than with the other values that are being printed, so there is no need to print the key.